pyro/distributions/multivariate_studentt.py

```diff
@@ -0,0 +1,95 @@
+import math
+
+import torch
+from torch.distributions import constraints
+from torch.distributions.utils import lazy_property
+
+from pyro.distributions.torch import Chi2
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.util import broadcast_shape
+
+
+class MultivariateStudentT(TorchDistribution):
+    """
+    Creates a multivariate Student's t-distribution parameterized by degree of
+    freedom :attr:`df`, mean :attr:`loc` and scale :attr:`scale_tril`.
+    :param ~torch.Tensor df: degrees of freedom
+    :param ~torch.Tensor loc: mean of the distribution
+    :param ~torch.Tensor scale_tril: scale of the distribution
+    """
+    arg_constraints = {'df': constraints.positive,
+                       'loc': constraints.real_vector,
+                       'scale_tril': constraints.lower_cholesky}
+    support = constraints.real_vector
+    has_rsample = True
+
+    def __init__(self, df, loc, scale_tril, validate_args=None):
+        dim = loc.size(-1)
+        assert scale_tril.shape[-2:] == (dim, dim)
+        batch_shape = broadcast_shape(df.shape, loc.shape[:-1], scale_tril.shape[:-2])
+        event_shape = (dim,)
+        self.df = df
```

Comment:
should `type(df)==float` also be allowed?

Reply:
Yes, it is. I’ll add support for it.